src/Sarif/NotYetAutoGenerated/ReportingDescriptorEqualityComparer.cs

```diff
@@ -0,0 +1,338 @@
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        public int GetHashCode(ReportingDescriptor obj)
+        {
+            if (ReferenceEquals(obj, null))
+            {
+                return 0;
+            }
+
+            int result = 17;
+            unchecked
+            {
+                if (obj.Id != null)
+                {
+                    result = (result * 31) + obj.Id.GetHashCode();
+                }
+
+                if (obj.DeprecatedIds != null)
+                {
+                    foreach (var value_4 in obj.DeprecatedIds)
+                    {
+                        result = result * 31;
+                        if (value_4 != null)
+                        {
+                            result = (result * 31) + value_4.GetHashCode();
+                        }
+                    }
+                }
+
+                if (obj.Guid != null)
+                {
+                    result = (result * 31) + obj.Guid.GetHashCode();
+                }
+
+                if (obj.DeprecatedGuids != null)
+                {
+                    foreach (var value_5 in obj.DeprecatedGuids)
+                    {
+                        result = result * 31;
+                        if (value_5 != null)
+                        {
+                            result = (result * 31) + value_5.GetHashCode();
+                        }
+                    }
+                }
+
+                if (obj.Name != null)
+                {
+                    result = (result * 31) + obj.Name.GetHashCode();
+                }
+
+                if (obj.DeprecatedNames != null)
+                {
+                    foreach (var value_6 in obj.DeprecatedNames)
+                    {
+                        result = result * 31;
+                        if (value_6 != null)
+                        {
+                            result = (result * 31) + value_6.GetHashCode();
+                        }
+                    }
+                }
+
+                if (obj.ShortDescription != null)
+                {
+                    result = (result * 31) + obj.ShortDescription.ValueGetHashCode();
+                }
+
+                if (obj.FullDescription != null)
+                {
+                    result = (result * 31) + obj.FullDescription.ValueGetHashCode();
+                }
+
+                if (obj.MessageStrings != null)
+                {
+                    // Use xor for dictionaries to be order-independent.
+                    int xor_0 = 0;
+                    foreach (var value_7 in obj.MessageStrings)
+                    {
+                        xor_0 ^= value_7.Key.GetHashCode();
+                        if (value_7.Value != null)
+                        {
+                            xor_0 ^= MultiformatMessageStringEqualityComparer
```

Comment:
>MultiformatMessageStringEqualityComparer [](start = 37, length = 40)

By chance, I found a serious problem in our equality comparer, which doesn't itself call the equality comparer for any dictionaries that contain values which are SARIF reference types (like this case, where we have a dictionary whose values are MultiformatMessageString instances. We probably have other exposure to this bug in the format, I will open an issue on it. For now, I've made a manual update to the autogenerated code to handle this case.